src/main/java/net/juniper/netconf/NetconfSession.java

```diff
@@ -450,13 +464,7 @@ public BufferedReader executeRPCRunning(Document rpcDoc) throws IOException {
      * @return Session ID as a string.
      */
     public String getSessionId() {
-        String[] split = serverCapability.split("<session-id>");
-        if (split.length != 2)
-            return null;
-        String[] idSplit = split[1].split("</session-id>");
-        if (idSplit.length != 2)
-            return null;
-        return idSplit[0];
+        return serverHello.getSessionId();
```

Comment:
just curious, could this be a static method in Hello.java?

Reply:
This couldn't be, no. An instance of the Hello object represents a single `<hello>` NETCONF element. So the session-id is an instance variable.

Reply:
👍